setup.cfg

```diff
@@ -14,4 +14,6 @@ ignore = E203, E501, E741, W503, W605
 max-line-length = 119
 
 [tool:pytest]
-doctest_optionflags=NUMBER NORMALIZE_WHITESPACE ELLIPSIS
\ No newline at end of file
+doctest_optionflags=NUMBER NORMALIZE_WHITESPACE ELLIPSIS
+filterwarnings =
+    error::FutureWarning:huggingface_hub* # Test fails if a FutureWarning is thrown by `huggingface_hub`
```

Comment:
I'm not super comfortable with having this in a setup.cfg, and thus have the CI suddenly errors once huggingface_hub issues a new deprecation warning. This should be tested in one of the daily runners IMO.

Reply:
I also feel the same, i.e. better not to in CircleCI for PRs. I didn't reply regarding this part, as I am not a fans of having it in daily CI neither: when a daily CI report suddenly contains a lot of warning failures, it is too much noisy to focus on the real issues we have to fix.

However, we have a way to collect warnings, and we can further filter them by the ones issued from `huggingface_hub`.
See for example

<img width="488" alt="Screenshot 2023-03-09 134910" src="https://user-images.githubusercontent.com/2521628/224027835-cafa9cc9-7b05-4948-9a9a-d1384a3542e2.png">

So, @Wauplin Could we do nothing regarding this, and I will open a PR to generate report regarding warnings from `huggingfface_hub`?

Reply:
@sgugger Yes sure, it was just a suggestion. @ydshieh that seems perfect yes. I'm removing it from my PR :)

Reply:
Removed in https://github.com/huggingface/transformers/pull/22051/commits/d95a6c5e673cb6a156330eecee289acf65ebcda6.

If you are fine with the PR, I let you review it and merge. Thanks! :)